astropy/io/ascii/core.py

```diff
@@ -578,7 +584,7 @@ class BaseHeader:
     """ None, int, or a function of ``lines`` that returns None or int """
     comment = None
     """ regular expression for comment lines """
-    splitter_class = DefaultSplitter
+    splitter_class: ClassVar[type[BaseSplitter]] = DefaultSplitter
```

Comment:
Also here I think we need to make sure that other attributes defined in this scope are (or aren't) `ClassVar`s